Sources/Service/Service.swift

```diff
@@ -0,0 +1,53 @@
+//
+//  Service.swift
+//  Cara
+//
+//  Created by Jelle Vandebeeck on 01/12/2018.
+//
+
+/// This it the main executor of requests.
+public class Service {
+    
+    // MARK: - Internal
+    
+    private let configuration: Configuration
+    private let networkService: NetworkService
+    
+    // MARK: - Init
+    
+    /// Initialize the Service layer.
+    ///
+    /// - parameter configuration: Configure the service layer through this instance.
+    init(configuration: Configuration) {
+        self.configuration = configuration
+        self.networkService = NetworkService(configuration: configuration)
+    }
+    
+    // MARK: - Execute
+    
+    /// Execute the given request and get an appropriate response returned.
+    ///
+    /// ```swift
+    /// let request = SomeRequest()
+    /// let serializer = SomeSerializer()
+    /// service.execute(request, with: serializer) { response in
```

Comment:
Zou serializer niet optional en default op `JSONSerializer` kunnen staan? Wordt dat niet het meest gebruikt?

Reply:
Ik zou het op dit niveau zo laten. Nu staat het nog vrij voor de gebruiker om zelf een `JSONRequest` te definiëren, of zelf een extensie te maken. Ik zou alles zo algemeen mogelijk proberen te houden in deze component 🤷‍♂️